Api/Controllers/CharacterTypesController.cs

```diff
@@ -0,0 +1,58 @@
+﻿using DomainLayer.Constants;
+using DomainLayer.Entities;
+using Microsoft.AspNetCore.Mvc;
+using RepositoryLayer.Repositories;
+using ServiceLayer.Business;
+using WebApiLayer.UserFeatures.Requests;
+
+namespace WebApiLayer.Controllers;
+
+[Route(Constants.HTTP.API_VERSION + "/gms/character-types")]
+public class CharacterTypesController : BaseController
+{
+    private readonly ICharacterTypeServices _characterTypeServices;
+    private readonly IGenericRepository<CharacterTypeEntity> _characterTypeRepo;
+    public CharacterTypesController(ICharacterTypeServices characterTypeServices, IGenericRepository<CharacterTypeEntity> characterTypeRepo)
+    {
+        _characterTypeServices = characterTypeServices;
+        _characterTypeRepo = characterTypeRepo;
+    }
+    [HttpGet]
+    public async Task<IActionResult> GetCharacterTypes()
+    {
+        var ctList = await _characterTypeServices.List();
+        return Ok(ctList);
+    }
+
+    [HttpGet("{id}")]
+    public async Task<IActionResult> GetCharacterType(Guid id)
+    {
+        var ctList = await _characterTypeServices.List();
+        return Ok(ctList);
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> Post([FromBody] CreateCharacterTypeRequest charType)
+    {
+        var newCT = new CharacterTypeEntity();
+        Mapper.Map(charType, newCT);
+        await _characterTypeServices.Create(newCT);
```

Comment:
do you sure that entity return by repo.Create is the same with your newCT? why  don't get the entity that return by repo.Create and response to client?